pkg/project/admission/nodeenv/admission.go

```diff
@@ -52,6 +57,16 @@ func (p *podNodeEnvironment) Admit(a admission.Attributes) (err error) {
 	if !p.cache.Running() {
 		return err
 	}
+
+	// if this is a pod being created by a daemonset we do not want to set the node selector automatically
+	isDaemonset, err := p.isDaemonsetPod(pod)
+	if err != nil {
+		return apierrors.NewForbidden(resource, name, fmt.Errorf("an unexpected error occured when checking for daemonsets: %v", err))
+	}
+	if isDaemonset {
+		return nil
```

Comment:
what prevents a user from faking the createdby annotation and circumventing the nodeSelector? Even a daemonset should be subject to a project's node selector